src/vanilla.ts

```diff
@@ -5,7 +5,11 @@ export type PartialState<T extends State> =
 export type StateSelector<T extends State, U> = (state: T) => U
 export type EqualityChecker<T> = (state: T, newState: unknown) => boolean
 export type StateListener<T> = (state: T) => void
```

Comment:
```suggestion
export type StateListener<T> = (state: T, previousState: T) => void
```

Would we like to change this too?